src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTests.cs

```diff
@@ -0,0 +1,317 @@
+            }
+            catch (XunitException)
+            {
+                // Ignore failures from delete-channel, its just a pre-cleanup that isn't really part of the test
+            }
+
+            await RunDarcAsync("add-channel", "--name", testChannelName, "--classification", "test").ConfigureAwait(false);
+
+            return AsyncDisposableValue.Create(testChannelName, async () =>
+            {
+                _output.WriteLine($"Cleaning up Test Channel {testChannelName}");
+                await RunDarcAsync("delete-channel", "--name", testChannelName).ConfigureAwait(false);
+            });
+        }
+
+        private async Task<AsyncDisposableValue<string>> CreateSubscriptionAsync(string sourceChannelName, string sourceRepo, string targetRepo, string targetBranch, string updateFrequency)
+        {
+            string output = await RunDarcAsync("add-subscription", "-q", "--no-trigger",
+                "--channel", sourceChannelName,
+                "--source-repo", GetRepoUrl("dotnet", sourceRepo),
+                "--target-repo", GetRepoUrl(targetRepo),
+                "--target-branch", targetBranch,
+                "--update-frequency", updateFrequency).ConfigureAwait(false);
+
+            Match match = Regex.Match(output, "Successfully created new subscription with id '([a-f0-9-]+)'");
+            if (match.Success)
+            {
+                string subscriptionId = match.Groups[1].Value;
+                return AsyncDisposableValue.Create(subscriptionId, async () =>
+                {
+                    _output.WriteLine($"Cleaning up Test Subscription {subscriptionId}");
+                    await RunDarcAsync("delete-subscriptions", "--id", subscriptionId, "--quiet").ConfigureAwait(false);
+                });
+            }
+
+            throw new XunitException("Unable to create subscription.");
+        }
+
+        private Task<int> CreateBuildAsync(string repositoryUrl, string branch, string commit, string buildNumber, IImmutableList<AssetData> assets)
+        {
+            return CreateBuildAsync(repositoryUrl, branch, commit, buildNumber, assets, ImmutableList<BuildRef>.Empty);
+        }
+
+        private async Task<int> CreateBuildAsync(string repositoryUrl, string branch, string commit, string buildNumber, IImmutableList<AssetData> assets, IImmutableList<BuildRef> dependencies)
+        {
+            Build build = await MaestroApi.Builds.CreateAsync(new BuildData(
+                commit: commit,
+                azureDevOpsAccount: "dnceng",
+                azureDevOpsProject: "internal",
+                azureDevOpsBuildNumber: buildNumber,
+                azureDevOpsRepository: repositoryUrl,
+                azureDevOpsBranch: branch,
+                publishUsingPipelines: false,
+                released: false)
+            {
+                AzureDevOpsBuildId = 144618,
+                AzureDevOpsBuildDefinitionId = 6,
```

Comment:
Are these default values? Anyway we can move these to a config?

Reply:
These were all hardcoded in the powershell version of these tests also. I'll throw them in the parameters.